File: 49.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string,vector<string>> hashMap;
        for(const string& str:strs){
            string sortedstr=str;
            sort(sortedstr.begin(),sortedstr.end());
            hashMap[sortedstr].push_back(str);
        }
        vector<vector<string>> result;
        for(auto& pair:hashMap){
            result.push_back(pair.second);
        }
        return result;

    }
};

File: 49.py
class Solution(object):
    def groupAnagrams(self, strs):
        hashMap=defaultdict(list)
        for str in strs:
            sortedStr=str
            sortedStr=''.join(sorted(str))
            hashMap[sortedStr].append(str)
        return list(hashMap.values())
        
